### DISCUSSION TITLE
Creating new content-type "Webinar"

### DISCUSSION THREAD

**Discussion by mahajanmahesh935** [Sunbird-Ed/Community#680] [Feature request]
Hi team,
Hope you are doing well!
Here we have came up with new approach. Where we are trying to create a new content-type "Webinar".
Course creator/Trainer Should be able create the webinar. Only registered users can join the webinar on scheduled time. This webinar can be added separately or can be part of a course. Based on the attendance, creator/trainer should also be able to provide the certification.
For this we have few questions we want help on:

- **What are the basic requirements for creating new content-type and what steps we have follow for same?**
- **Any configurations should be updated?** If yes please guide us about it
- **What will be code changes we need to do for creating new content-type?**

We appreciate any insights. Thank you!

@vinukumar-vs ,@rajeevsathish, @AmiableAnil , @Krishnaj20, @chitranshu-keshav, @SanthoshVasabhaktula, @gandham-santhosh , @coolbung @komalm @manishakapadnis

**Comment by mahajanmahesh935** [comment]
Hi team,
Your insights and assistance would be highly appreciated in resolving these matters. Looking forward to your response.

**Reply by mohitga** [reply]
Hi @mahajanmahesh935  

Creating webinar new content type can we done. You need to plan the workflow like 
1. when to publish the webinar
2. Who all can join the webinar 
3. What is life cycle of the webinar
4. What telemetry is captured 
5. How the certification generation is stitched with webinar attendance. 

you can use [Ask dot ](https://bot.sunbird.org/ )to ask questions, same is added to the Ed.sunbird.org website 
<img width="1407" alt="image" src="https://github.com/Sunbird-Ed/Community/assets/22144853/077375bb-3bae-4061-9ebf-ce5ecafe5465">

Question : 

> How can i create "Webinar" as new content type in Sunbird ED


Answer : 

> To create a new content type "Webinar" in Sunbird ED, you will need to follow these steps:
> 
> Content Type Registration: Register the new content type "Webinar" with Sunbird ED by defining its metadata, attributes, and functionalities. This can be done by accessing the Sunbird ED's backend and modifying the content type registration configuration files.
> 
> Content Player Integration: Integrate a content player specifically designed for webinars. This player should support the playback of live or recorded webinars, with features like video streaming, chat functionality, and presentation sharing. The player should be compatible with Sunbird ED's infrastructure.
> 
> Content Creation: Develop a content creation interface within Sunbird ED's authoring tool that allows content creators to create and upload webinar content. This interface should facilitate the inclusion of webinar-specific attributes such as title, description, presenter details, date, time, and duration.
> 
> Content Management: Incorporate the necessary content management functionalities for webinars, including options to schedule webinars, manage recordings, and control access permissions.
> 
> User Engagement: Integrate features that encourage user engagement with webinars, such as the ability to submit questions, participate in discussions, and provide feedback.
> 
> Testing and Deployment: Thoroughly test the webinar content type and player integration to ensure they are working correctly. Once validated, deploy the changes to the Sunbird ED production environment.
> 
> Please note that creating a new content type in Sunbird ED may require technical expertise and access to the Sunbird ED's backend. It is recommended to consult the Sunbird documentation or seek assistance from the Sunbird community for detailed guidance on creating custom content types.

 
 @vinukumar-vs  @AmiableAnil can help you guide you with further queries

**Reply by paritshivani** [reply]
Thanks, @mohitga 
We have one content types as events
Can we use same for webinar??

**Comment by vinukumar-vs** [comment]
@paritshivani I would suggest you to check/confirm do you want to create the new content-type or PrimaryCategory? 

Today we have content-types like "content", "collection", "question-set" etc. This is baked in the code level. If you introduce new content-type, you have to make respective service changes to handle it.

I would suggest to use new primaryCategory as "webinar" of type collection. 
Refer the API collection and knowlg-friday video sessions.

Postman collection 
https://www.postman.com/sunbird-building-blocks/workspace/sunbird-ed/collection/25186239-6547f9fc-5f16-495c-b5e0-d76f5a78d66d

Knowlg-Friday video session
https://github.com/orgs/Sunbird-Knowlg/discussions?discussions_q=is%3Aopen+friday

Refer to this session:
https://github.com/orgs/Sunbird-Knowlg/discussions/20#discussioncomment-2970492

**Comment by anisha147** [comment]
Hi @paritshivani 

We hope the suggestions shared by @vinukumar-vs helped you resolve your queries. If yes, could you please close this thread from your end?

cc: @vinukumar-vs
